Generate new HTTP Python Connector doesn't install all libs

<!-- 
Welcome to Airbyte!
We're really appreciate your report and know that this will help us build an 
amazing tool. If you want to contribute yourself, you can find a good place 
to start by searching for the good-first-issues label or maybe... by trying 
to solve this one? (we can help debug this with you!)

Right now we are in alpha, so we're releasing versions a lot more frequently than 
normal. You can help us get to the root of the problem faster by filling out the
questionnaire below!

It's really important having all information and context. 
You can remove the examples bellow and fill out with your information.

--> 

## Enviroment
- **Airbyte version**:  0.29.21-alpha
- **OS Version / Instance**: macOS


## Current Behavior
I'm trying to create a new connector (HTTP Python) and execute commands from README file.
```
file /Users/marcosmarx/airbyte/airbyte-integrations/connectors/source-python-http-custom/unit_tests/test_streams.py, line 99
  def test_backoff_time(patch_base_class):
file /Users/marcosmarx/airbyte/airbyte-integrations/connectors/source-python-http-custom/unit_tests/test_streams.py, line 32
  @pytest.fixture
  def patch_base_class(mocker):
E       fixture 'mocker' not found
>       available fixtures: cache, capfd, capfdbinary, caplog, capsys, capsysbinary, doctest_namespace, monkeypatch, patch_base_class, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory
>       use 'pytest --fixtures [testpath]' for help on them.

/Users/marcosmarx/airbyte/airbyte-integrations/connectors/source-python-http-custom/unit_tests/test_streams.py:32
```

See [Slack convo](https://airbytehq.slack.com/archives/C01VDDEGL7M/p1632406270447800)

## Expected Behavior
The instructions must work.

## Logs
*If applicable, please upload the logs from the failing operation. 
For sync jobs, you can download the full logs from the UI by going to the sync attempt page and 
clicking the download logs button at the top right of the logs display window.*

<details>
<summary>LOG</summary> 

```

replace this with
your long log
output here

```

</details>

## Steps to Reproduce
1.
2.
3.

## Are you willing to submit a PR?
<!--- 
We accept contributions! 
Don't feel pressured, but if you want to contribute we can help you by giving some tips, 
highlighting the necessary code change or explaining any relevant point your feature will impact.
You can also send questions on #dev Slack channel.

We understand if you can't submit a PR and we're tremendously grateful
that you've already contributed by suggesting a new feature.
-->
Remove this with your answer.

--- a/airbyte-integrations/connector-templates/source-python-http-api/README.md.hbs
+++ b/airbyte-integrations/connector-templates/source-python-http-api/README.md.hbs
@@ -21,6 +21,7 @@ development environment of choice. To activate it from the terminal, run:
 ```
 source .venv/bin/activate
 pip install -r requirements.txt
+pip install '.[tests]'
 ```
 If you are in an IDE, follow your IDE's instructions to activate the virtualenv.
 

--- a/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_incremental_streams.py.hbs
+++ b/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_incremental_streams.py.hbs
@@ -66,9 +66,11 @@ def test_supports_incremental(patch_incremental_base_class, mocker):
     stream = Incremental{{properCase name}}Stream()
     assert stream.supports_incremental
 
+
 def test_source_defined_cursor(patch_incremental_base_class):
     stream = Incremental{{properCase name}}Stream()
     assert stream.source_defined_cursor
+
 
 def test_stream_checkpoint_interval(patch_incremental_base_class):
     stream = Incremental{{properCase name}}Stream()

--- a/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_incremental_streams.py.hbs
+++ b/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_incremental_streams.py.hbs
@@ -23,7 +23,6 @@
 #
 
 from pytest import fixture
-from unittest.mock import MagicMock
 
 from airbyte_cdk.models import SyncMode
 from source_{{snakeCase name}}.source import Incremental{{properCase name}}Stream
@@ -46,7 +45,6 @@ def test_cursor_field(patch_incremental_base_class):
 
 def test_get_updated_state(patch_incremental_base_class):
     stream = Incremental{{properCase name}}Stream()
-    expected_cursor_field = []
     # TODO: replace this with your input parameters
     inputs = {"current_stream_state": None, "latest_record": None}
     # TODO: replace this with your expected updated stream state
@@ -56,7 +54,6 @@ def test_get_updated_state(patch_incremental_base_class):
 
 def test_stream_slices(patch_incremental_base_class):
     stream = Incremental{{properCase name}}Stream()
-    expected_cursor_field = []
     # TODO: replace this with your input parameters
     inputs = {"sync_mode": SyncMode.incremental, "cursor_field": [], "stream_state": {}}
     # TODO: replace this with your expected stream slices list

--- a/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_source.py.hbs
+++ b/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_source.py.hbs
@@ -31,6 +31,7 @@ def test_check_connection(mocker):
     logger_mock, config_mock = MagicMock(), MagicMock()
     assert source.check_connection(logger_mock, config_mock) == (True, None)
 
+
 def test_streams(mocker):
     source = Source{{properCase name}}()
     config_mock = MagicMock()

--- a/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_source.py.hbs
+++ b/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_source.py.hbs
@@ -22,8 +22,9 @@
 # SOFTWARE.
 #
 
-from source_{{snakeCase name}}.source import Source{{properCase name}}
 from unittest.mock import MagicMock
+
+from source_{{snakeCase name}}.source import Source{{properCase name}}
 
 
 def test_check_connection(mocker):

--- a/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_streams.py.hbs
+++ b/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_streams.py.hbs
@@ -24,8 +24,8 @@
 
 from http import HTTPStatus
 from unittest.mock import MagicMock
-import pytest
 
+import pytest
 from source_{{snakeCase name}}.source import {{properCase name}}Stream
 
 

--- a/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_streams.py.hbs
+++ b/airbyte-integrations/connector-templates/source-python-http-api/unit_tests/test_streams.py.hbs
@@ -70,7 +70,7 @@ def test_request_headers(patch_base_class):
     inputs = {"stream_slice": None, "stream_state": None, "next_page_token": None}
     # TODO: replace this with your expected request headers
     expected_headers = {}
-    assert stream.request_headers(**inputs) == {}
+    assert stream.request_headers(**inputs) == expected_headers
 
 
 def test_http_method(patch_base_class):

--- a/airbyte-integrations/connectors/source-scaffold-source-http/README.md
+++ b/airbyte-integrations/connectors/source-scaffold-source-http/README.md
@@ -21,6 +21,7 @@ development environment of choice. To activate it from the terminal, run:
 ```
 source .venv/bin/activate
 pip install -r requirements.txt
+pip install '.[tests]'
 ```
 If you are in an IDE, follow your IDE's instructions to activate the virtualenv.
 

--- a/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/__init__.py
+++ b/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/__init__.py
@@ -1,0 +1,23 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#

--- a/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_incremental_streams.py
+++ b/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_incremental_streams.py
@@ -22,10 +22,8 @@
 # SOFTWARE.
 #
 
-from pytest import fixture
-from unittest.mock import MagicMock
-
 from airbyte_cdk.models import SyncMode
+from pytest import fixture
 from source_scaffold_source_http.source import IncrementalScaffoldSourceHttpStream
 
 
@@ -46,7 +44,6 @@ def test_cursor_field(patch_incremental_base_class):
 
 def test_get_updated_state(patch_incremental_base_class):
     stream = IncrementalScaffoldSourceHttpStream()
-    expected_cursor_field = []
     # TODO: replace this with your input parameters
     inputs = {"current_stream_state": None, "latest_record": None}
     # TODO: replace this with your expected updated stream state
@@ -56,7 +53,6 @@ def test_get_updated_state(patch_incremental_base_class):
 
 def test_stream_slices(patch_incremental_base_class):
     stream = IncrementalScaffoldSourceHttpStream()
-    expected_cursor_field = []
     # TODO: replace this with your input parameters
     inputs = {"sync_mode": SyncMode.incremental, "cursor_field": [], "stream_state": {}}
     # TODO: replace this with your expected stream slices list
@@ -69,9 +65,11 @@ def test_supports_incremental(patch_incremental_base_class, mocker):
     stream = IncrementalScaffoldSourceHttpStream()
     assert stream.supports_incremental
 
+
 def test_source_defined_cursor(patch_incremental_base_class):
     stream = IncrementalScaffoldSourceHttpStream()
     assert stream.source_defined_cursor
+
 
 def test_stream_checkpoint_interval(patch_incremental_base_class):
     stream = IncrementalScaffoldSourceHttpStream()

--- a/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_incremental_streams.py
+++ b/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_incremental_streams.py
@@ -1,0 +1,80 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from pytest import fixture
+from unittest.mock import MagicMock
+
+from airbyte_cdk.models import SyncMode
+from source_scaffold_source_http.source import IncrementalScaffoldSourceHttpStream
+
+
+@fixture
+def patch_incremental_base_class(mocker):
+    # Mock abstract methods to enable instantiating abstract class
+    mocker.patch.object(IncrementalScaffoldSourceHttpStream, "path", "v0/example_endpoint")
+    mocker.patch.object(IncrementalScaffoldSourceHttpStream, "primary_key", "test_primary_key")
+    mocker.patch.object(IncrementalScaffoldSourceHttpStream, "__abstractmethods__", set())
+
+
+def test_cursor_field(patch_incremental_base_class):
+    stream = IncrementalScaffoldSourceHttpStream()
+    # TODO: replace this with your expected cursor field
+    expected_cursor_field = []
+    assert stream.cursor_field == expected_cursor_field
+
+
+def test_get_updated_state(patch_incremental_base_class):
+    stream = IncrementalScaffoldSourceHttpStream()
+    expected_cursor_field = []
+    # TODO: replace this with your input parameters
+    inputs = {"current_stream_state": None, "latest_record": None}
+    # TODO: replace this with your expected updated stream state
+    expected_state = {}
+    assert stream.get_updated_state(**inputs) == expected_state
+
+
+def test_stream_slices(patch_incremental_base_class):
+    stream = IncrementalScaffoldSourceHttpStream()
+    expected_cursor_field = []
+    # TODO: replace this with your input parameters
+    inputs = {"sync_mode": SyncMode.incremental, "cursor_field": [], "stream_state": {}}
+    # TODO: replace this with your expected stream slices list
+    expected_stream_slice = [None]
+    assert stream.stream_slices(**inputs) == expected_stream_slice
+
+
+def test_supports_incremental(patch_incremental_base_class, mocker):
+    mocker.patch.object(IncrementalScaffoldSourceHttpStream, "cursor_field", "dummy_field")
+    stream = IncrementalScaffoldSourceHttpStream()
+    assert stream.supports_incremental
+
+def test_source_defined_cursor(patch_incremental_base_class):
+    stream = IncrementalScaffoldSourceHttpStream()
+    assert stream.source_defined_cursor
+
+def test_stream_checkpoint_interval(patch_incremental_base_class):
+    stream = IncrementalScaffoldSourceHttpStream()
+    # TODO: replace this with your expected checkpoint interval
+    expected_checkpoint_interval = None
+    assert stream.state_checkpoint_interval == expected_checkpoint_interval

--- a/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_incremental_streams.py
+++ b/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_incremental_streams.py
@@ -22,8 +22,9 @@
 # SOFTWARE.
 #
 
-from airbyte_cdk.models import SyncMode
 from pytest import fixture
+
+from airbyte_cdk.models import SyncMode
 from source_scaffold_source_http.source import IncrementalScaffoldSourceHttpStream
 
 

--- a/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_source.py
+++ b/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_source.py
@@ -22,14 +22,16 @@
 # SOFTWARE.
 #
 
-from source_scaffold_source_http.source import SourceScaffoldSourceHttp
 from unittest.mock import MagicMock
+
+from source_scaffold_source_http.source import SourceScaffoldSourceHttp
 
 
 def test_check_connection(mocker):
     source = SourceScaffoldSourceHttp()
     logger_mock, config_mock = MagicMock(), MagicMock()
     assert source.check_connection(logger_mock, config_mock) == (True, None)
+
 
 def test_streams(mocker):
     source = SourceScaffoldSourceHttp()

--- a/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_source.py
+++ b/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_source.py
@@ -1,0 +1,40 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from source_scaffold_source_http.source import SourceScaffoldSourceHttp
+from unittest.mock import MagicMock
+
+
+def test_check_connection(mocker):
+    source = SourceScaffoldSourceHttp()
+    logger_mock, config_mock = MagicMock(), MagicMock()
+    assert source.check_connection(logger_mock, config_mock) == (True, None)
+
+def test_streams(mocker):
+    source = SourceScaffoldSourceHttp()
+    config_mock = MagicMock()
+    streams = source.streams(config_mock)
+    # TODO: replace this with your streams number
+    expected_streams_number = 2
+    assert len(streams) == expected_streams_number

--- a/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_streams.py
+++ b/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_streams.py
@@ -1,0 +1,103 @@
+#
+# MIT License
+#
+# Copyright (c) 2020 Airbyte
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+from http import HTTPStatus
+from unittest.mock import MagicMock
+import pytest
+
+from source_scaffold_source_http.source import ScaffoldSourceHttpStream
+
+
+@pytest.fixture
+def patch_base_class(mocker):
+    # Mock abstract methods to enable instantiating abstract class
+    mocker.patch.object(ScaffoldSourceHttpStream, "path", "v0/example_endpoint")
+    mocker.patch.object(ScaffoldSourceHttpStream, "primary_key", "test_primary_key")
+    mocker.patch.object(ScaffoldSourceHttpStream, "__abstractmethods__", set())
+
+
+def test_request_params(patch_base_class):
+    stream = ScaffoldSourceHttpStream()
+    # TODO: replace this with your input parameters
+    inputs = {"stream_slice": None, "stream_state": None, "next_page_token": None}
+    # TODO: replace this with your expected request parameters
+    expected_params = {}
+    assert stream.request_params(**inputs) == expected_params
+
+
+def test_next_page_token(patch_base_class):
+    stream = ScaffoldSourceHttpStream()
+    # TODO: replace this with your input parameters
+    inputs = {"response": MagicMock()}
+    # TODO: replace this with your expected next page token
+    expected_token = None
+    assert stream.next_page_token(**inputs) == expected_token
+
+
+def test_parse_response(patch_base_class):
+    stream = ScaffoldSourceHttpStream()
+    # TODO: replace this with your input parameters
+    inputs = {"response": MagicMock()}
+    # TODO: replace this with your expected parced object
+    expected_parsed_object = {}
+    assert next(stream.parse_response(**inputs)) == expected_parsed_object
+
+
+def test_request_headers(patch_base_class):
+    stream = ScaffoldSourceHttpStream()
+    # TODO: replace this with your input parameters
+    inputs = {"stream_slice": None, "stream_state": None, "next_page_token": None}
+    # TODO: replace this with your expected request headers
+    expected_headers = {}
+    assert stream.request_headers(**inputs) == {}
+
+
+def test_http_method(patch_base_class):
+    stream = ScaffoldSourceHttpStream()
+    # TODO: replace this with your expected http request method
+    expected_method = "GET"
+    assert stream.http_method == expected_method
+
+
+@pytest.mark.parametrize(
+    ("http_status", "should_retry"),
+    [
+        (HTTPStatus.OK, False),
+        (HTTPStatus.BAD_REQUEST, False),
+        (HTTPStatus.TOO_MANY_REQUESTS, True),
+        (HTTPStatus.INTERNAL_SERVER_ERROR, True),
+    ],
+)
+def test_should_retry(patch_base_class, http_status, should_retry):
+    response_mock = MagicMock()
+    response_mock.status_code = http_status
+    stream = ScaffoldSourceHttpStream()
+    assert stream.should_retry(response_mock) == should_retry
+
+
+def test_backoff_time(patch_base_class):
+    response_mock = MagicMock()
+    stream = ScaffoldSourceHttpStream()
+    expected_backoff_time = None
+    assert stream.backoff_time(response_mock) == expected_backoff_time

--- a/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_streams.py
+++ b/airbyte-integrations/connectors/source-scaffold-source-http/unit_tests/test_streams.py
@@ -24,8 +24,8 @@
 
 from http import HTTPStatus
 from unittest.mock import MagicMock
-import pytest
 
+import pytest
 from source_scaffold_source_http.source import ScaffoldSourceHttpStream
 
 
@@ -70,7 +70,7 @@ def test_request_headers(patch_base_class):
     inputs = {"stream_slice": None, "stream_state": None, "next_page_token": None}
     # TODO: replace this with your expected request headers
     expected_headers = {}
-    assert stream.request_headers(**inputs) == {}
+    assert stream.request_headers(**inputs) == expected_headers
 
 
 def test_http_method(patch_base_class):

